Add a Few Codegen Tests

veera-sivarajan opened the pull request:
Closes #86109
Closes #64219

Those issues somehow got fixed over time. 

So, this PR adds a couple of codegen tests to ensure we don't regress in the future.

veera-sivarajan committed:
Add a Few Codegen Tests

jieyouxu commented on tests/codegen/issues/issue-86109.rs:
```diff
@@ -0,0 +1,25 @@
+//@ compile-flags: -O
+//! Test for https://github.com/rust-lang/rust/issues/86109
+//! Check LLVM can eliminate the impossible division by zero check by
+//! ensuring there is no call (to panic) instruction.
```
Remark: that the division by zero was present in 1.69 but disappeared in 1.70.

jieyouxu reviewed (changes requested):
The tests themselves look good, and the comments are good, but please also:

1. Give the tests meaningful names (not just issue-xxxxx). E.g. `fn-ptr-never-ret-ty-has-noreturn-attr.rs`.
2. Please move them into meaningful subdirectories instead of `issues/`.

Other than that, LGTM.

jieyouxu commented:
@rustbot author

veera-sivarajan commented:
Thank you for the feedback.

I've renamed the tests and added the remark. But I cannot find a more appropriate directory for these tests than `codegen/issues`.

jieyouxu commented:
That's fine. I'll r+ after PR CI is green.

workingjubilee commented:
@bors r=jieyouxu

bors commented:
:pushpin: Commit 85d6d9c8080ac315fa37259013d9c2db6a34e6f1 has been approved by `jieyouxu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jieyouxu 85d6d9c8080ac315fa37259013d9c2db6a34e6f1 -->
<!-- homu: {"type":"Approved","sha":"85d6d9c8080ac315fa37259013d9c2db6a34e6f1","approver":"jieyouxu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

workingjubilee commented:
did not pass: https://github.com/rust-lang/rust/pull/132211#issuecomment-2439822384

@bors r-

jieyouxu commented:
Yeah, that matched on a `tail call` on aarc64 apple

jieyouxu commented:
Thanks!
@bors r+ rollup=iffy
